src/devices/Amg88xx/Amg88xx.cs

```diff
@@ -0,0 +1,475 @@
+            // only the bit to be cleared is set, the other bits need to be 0
+            SetRegister(Register.SCLR, 1 << (byte)StatusClearBit.OVFCLR);
+        }
+
+        /// <summary>
+        /// Gets the thermistor overflow flag from the status register.
+        /// Note: the bit is only menthioned in early versions of the reference specification.
+        /// It is not clear whether this is a specification error or a change in a newer
+        /// revision of the sensor.
+        /// </summary>
+        /// <returns>Thermistor overflow flag</returns>
+        public bool HasThermistorOverflow()
+        {
+            return GetBit(Register.STAT, (byte)StatusFlagBit.OVF_THS);
+        }
+
+        /// <summary>
+        /// Clears the thermistor overflow flag in the status register
+        /// </summary>
+        public void ClearThermistorOverflow()
+        {
+            // only the bit to be cleared is set, the other bits need to be 0
+            SetRegister(Register.SCLR, 1 << (byte)StatusClearBit.OVFTHCLR);
+        }
+
+        /// <summary>
+        /// Gets the interrupt flag from the status register
+        /// </summary>
+        /// <returns>Interrupt flag</returns>
+        public bool HasInterrupt()
+        {
+            return GetBit(Register.STAT, (byte)StatusFlagBit.INTF);
+        }
+
+        /// <summary>
+        /// Clears the interrupt flag in the status register
+        /// </summary>
+        public void ClearInterrupt()
+        {
+            // only the bit to be cleared is set, the other bits need to be 0
+            SetRegister(Register.SCLR, 1 << (byte)StatusClearBit.INTCLR);
+        }
+
+        /// <summary>
+        /// Clears all flags in the status register
+        /// </summary>
+        public void ClearAllFlags()
+        {
+            // only the bit to be cleared is set, the other bits need to be 0
+            SetRegister(Register.SCLR, (1 << (byte)StatusClearBit.OVFCLR) | (1 << (byte)StatusClearBit.OVFTHCLR) | (1 << (byte)StatusClearBit.INTCLR));
+        }
+
+        #endregion
+
+        #region Moving average
+
+        /// <summary>
+        /// Get the state of the moving average mode
+        /// </summary>
+        /// <returns>True, if moving average mode is on</returns>
+        public bool GetMovingAverageModeState()
+        {
+            return GetBit(Register.AVE, (byte)MovingAverageModeBit.MAMOD);
+        }
+
+        /// <summary>
+        /// Sets the moving average mode state.
+        /// </summary>
+        /// <param name="state">True, to switch moving average on</param>
+        public void SetMovingAverageModeState(bool state)
+        {
+            SetBit(Register.AVE, (byte)MovingAverageModeBit.MAMOD, state);
+        }
+
+        #endregion
+
+        #region Frame Rate
+
+        /// <summary>
+        /// Get the current frame rate.
+        /// </summary>
+        /// <returns>Frame rate (either 1 or 10fps) </returns>
+        public int GetFrameRate()
```

Comment:
@Ellerbach: For the sake of consistency the the moving average I made this a Get-/Set-API. Would you change this to a property even if MovingAverageModeState would be a Get-/Set-API?

Reply:
This one seems closer to a property. At least my understanding. Now is there are only 2 choices for the framerate, why not having a public enum for the 2 possible options? 1 and 10 fps?

Reply:
@Ellerbach Yes, I do generally agree that this could be a property. I was just thinking about consistency. As Get/SetMovingAverageModeState can't be a property I thought it would be "cleaner" to keep all a discrete getter/setter combinations. However, the majority of functions could be expressed as a property.
So I leave it your choice - consistency or properties where possible?

Reply:
If you accept only few values (i.e. 1 or 10) consider creating an enum (Amg88xxFrameRate or similar)

Reply:
(you should still check for OutOfRange though)

Reply:
I changed this to a property, too.